packages/template-retail-react-app/app/components/store-locator-modal/store-locator-content.jsx

```diff
@@ -58,17 +176,62 @@ const StoreLocatorContent = ({form, submitForm, storesInfo, searchStoresParams})
                                   id: 'store_locator.description.no_locations',
                                   defaultMessage: 'Sorry, there are no locations in this area'
                               })
-                            : intl.formatMessage(
+                            : searchStoresParams.postalCode !== undefined
+                            ? `${intl.formatMessage(
                                   {
-                                      id: 'store_locator.description.viewing_within',
-                                      defaultMessage: 'Viewing stores within {distance} miles'
+                                      id: 'store_locator.description.viewing_near_postal_code',
+                                      defaultMessage:
+                                          'Viewing stores within {distance}{distanceUnit} of {postalCode} in '
                                   },
-                                  {distance: STORE_LOCATOR_DISTANCE}
+                                  {
+                                      distance: STORE_LOCATOR_DISTANCE,
+                                      distanceUnit: STORE_LOCATOR_DISTANCE_UNIT,
+                                      postalCode: searchStoresParams.postalCode
+                                  }
                               )}
+                            ${
+                                SUPPORTED_STORE_LOCATOR_COUNTRIES.length !== 0
+                                    ? intl.formatMessage(
+                                          SUPPORTED_STORE_LOCATOR_COUNTRIES.find(
+                                              (o) =>
+                                                  o.countryCode === searchStoresParams.countryCode
+                                          ).countryName
+                                      )
+                                    : intl.formatMessage(DEFAULT_STORE_LOCATOR_COUNTRY.countryName)
+                            }`
+                            : intl.formatMessage({
+                                  id: 'store_locator.description.viewing_near_your_location',
+                                  defaultMessage: 'Viewing stores near your location'
+                              })}
```

Comment:
I think it's not clear at first glance what this block is doing. I think the way the code is laid out with multiple colons (multiple if-else and even some nested-if) is making it harder to read. Can we refactor it so it's clearer pls?